File: FlaskServer/controllers/InterestedPersonController.py
from flask import Blueprint, jsonify, make_response, request
from http import HTTPStatus
import logging
from services.InterestedPersonService import InterestedPersonService

interested_person_bp = Blueprint('interested_person', __name__, url_prefix='/interested-person')


@interested_person_bp.route('/get-all')
def get_all():
    service = InterestedPersonService()
    try:
        result = service.get_all()
        return make_response(jsonify(result), HTTPStatus.OK)
    except Exception as e:
        logging.error("Error fetching interested persons", e)
        return make_response(jsonify({'error': str(e)}), HTTPStatus.INTERNAL_SERVER_ERROR)


@interested_person_bp.route('/update', methods=['PUT'])
def update():
    try:
        data = request.get_json()
        if not data or 'id' not in data:
            return make_response(jsonify({'error': 'Missing ID or request body'}), HTTPStatus.BAD_REQUEST)

        service = InterestedPersonService()
        updated_person = service.update(data)

        if updated_person is None:
            return make_response(jsonify({'error': 'Person not found'}), HTTPStatus.NOT_FOUND)

        return make_response(jsonify(updated_person.get_dict()), HTTPStatus.OK)

    except Exception as e:
        logging.exception("Error updating interested person")
        return make_response(jsonify({'error': str(e)}), HTTPStatus.INTERNAL_SERVER_ERROR)

File: evangelism-assistant-server/src/main/kotlin/com/quenazapps/evangelism_assistant/controller/InterestedPersonController.kt
package com.quenazapps.evangelism_assistant.controller

import com.quenazapps.evangelism_assistant.service.InterestedPersonService
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import org.slf4j.LoggerFactory
import org.slf4j.Logger
import org.springframework.web.bind.annotation.PutMapping
import org.springframework.web.bind.annotation.RequestBody

@RestController
@RequestMapping("/interested-person")
class InterestedPersonController(
    private val interestedPersonService: InterestedPersonService
) {
    companion object {
        private val log: Logger = LoggerFactory.getLogger(InterestedPersonController::class.java)
    }

    @GetMapping("/get-all")
    fun getBetweenReferenceIds(): ResponseEntity<Any> {
        return try {
            val people = interestedPersonService.findAllDTOs()
            ResponseEntity.status(HttpStatus.OK).body(people)
        } catch (e: Exception) {
            log.error(e.message, e)
            ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(
                mapOf("status" to HttpStatus.INTERNAL_SERVER_ERROR.value(), "message" to e.message)
            )
        }
    }

    @PutMapping("/update")
    fun update(@RequestBody data: Map<String, Any>): ResponseEntity<Any> {
        return try {

            val updated = interestedPersonService.update(data)

            if (updated == null) {
                return ResponseEntity.status(HttpStatus.NOT_FOUND).body(
                    mapOf("error" to "Person not found")
                )
            }

            ResponseEntity.ok(updated.toDTO())

        } catch (e: Exception) {
            log.error("Error updating interested person", e)
            ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(
                mapOf("error" to e.message)
            )
        }
    }
}